File: coding_test/stack_queue/2_printer.py
def solution(priorities, location):
    
    ziped = [(x,y) for x, y in enumerate(priorities)]
    
    printed = []
    
    while len(ziped) != 0:
        a = [x[1] for x in ziped];
        b = ziped.pop(0)
        
        if max(a) == b[1]:
            printed.append(b)
        else:
            ziped.append(b)

    for i, z in enumerate(printed):
        if z[0] == location:
            return i+1
    answer = 0
    return answer

File: coding_test/stack_queue/2_printer.js
function solution(priorities, location) {
    
    count = 0
    pair = priorities.map((x, i) => [x,i] )
    
    while (pair.length != 0) {
        x = pair.shift()
        if (x[0] < Math.max(...pair.map(x => x[0]))) {
            pair.push(x)
        } else {
            count += 1
            if (x[1] === location)
                return count
        }
    }
    
    var answer = 0;
    return answer;
}